lib/routemaster/responses/hateoas_response.rb

```diff
@@ -13,6 +13,27 @@ class HateoasResponse
       attr_reader :response
       def_delegators :@response, :body, :status, :headers, :success?
 
+      class << self
+        def build(response, client: nil)
+          if contains_a_list_with_the_same_key_as_the_path?(response)
+            EnumerableHateoasResponse.new(response, resource_name(response), client: client)
+          else
+            HateoasResponse.new(response, client: client)
+          end
+        end
```

Comment:
It's not very clear what the requirements are in order to use `EnumerableHateoasResponse` rather than the normal response.

I would probably change the method name defined on line 27 to describe the scope of the method rather than its implementation, i.e. `index?(response)` or `is_a_collection?(response)` or something along those lines.

Thoughts?